lib/src/main/java/dev/openfeature/javasdk/Hook.java

```diff
@@ -3,6 +3,7 @@
 import com.google.common.collect.ImmutableMap;
 
 // TODO: interface? or abstract class?
+// I suppose an interface with no-op default methods is slightly more flexible, since you can implement multiple interfaces?
```

Comment:
Love that! I didn't know about this feature. https://github.com/open-feature/java-sdk/issues/8

Reply:
Ya I think the only downsides are that it's possible for the methods to clash with those of other interfaces, and that the default impls cant reference any state (obviously, since interfaces don't have any).